src/pas/plugins/oidc/services/oidc/oidc.py

```diff
@@ -0,0 +1,235 @@
+        except OAuth2ConnectionException:
+            self.request.response.setStatus(500)
+            return {
+                "error": {
+                    "type": "Configuration error",
+                    "message": _("Provider is not properly configured."),
+                }
+            }
+        try:
+            auth_req = client.construct_AuthorizationRequest(request_args=args)
+            login_url = auth_req.request(client.authorization_endpoint)
+        except Exception as e:
+            logger.error(e)
+            self.request.response.setStatus(500)
+            return {
+                "error": {
+                    "type": "Runtime error",
+                    "message": _(
+                        "There was an error during the login process. Please try again."
+                    ),
+                }
+            }
+        else:
+            return {
+                "next_url": login_url,
+                "came_from": session.get("came_from"),
+            }
+
+
+class LogoutGet(LoginOIDC):
+    """Logout a user."""
+
+    def reply(self) -> dict:
+        """Generate URL and session information to be used by the frontend.
+
+        :returns: URL and session information.
+        """
+        provider = "oidc"
+        plugin = self.plugin
+        if not (plugin and provider == "oidc"):
+            return self._provider_not_found(provider)
+
+        try:
+            client = plugin.get_oauth2_client()
+        except OAuth2ConnectionException:
+            self.request.response.setStatus(500)
+            return {
+                "error": {
+                    "type": "Configuration error",
+                    "message": _("Provider is not properly configured."),
+                }
+            }
+        redirect_uri = utils.url_cleanup(api.portal.get().absolute_url())
+
+        if plugin.getProperty("use_deprecated_redirect_uri_for_logout"):
+            args = {
+                "redirect_uri": redirect_uri,
+            }
+        else:
+            args = {
+                "post_logout_redirect_uri": redirect_uri,
+                "client_id": plugin.getProperty("client_id"),
+            }
+
+        pas = api.portal.get_tool("acl_users")
+        auth_cookie_name = pas.credentials_cookie_auth.cookie_name
+
+        # end_req = client.construct_EndSessionRequest(request_args=args)
+        end_req = EndSessionRequest(**args)
+        logout_url = end_req.request(client.end_session_endpoint)
+        self.request.response.expireCookie(auth_cookie_name, path="/")
+        self.request.response.expireCookie("auth_token", path="/")
+        return {
+            "next_url": logout_url,
+            "came_from": redirect_uri,
+        }
+
+
+class Post(LoginOIDC):
+    """Handles OIDC login and returns a JSON web token (JWT)."""
+
+    def check_permission(self) -> bool:
+        return True
+
+    def _annotate_transaction(self, action: str, user: MemberData):
```

Comment:
This is useful for ?

Reply:
As Anonymous users use this service, I bypass the check for "plone.restapi: Use REST API" permission